### ISSUE TITLE
Issue: Setup & Core Interfaces

### ISSUE THREAD

**Issue by M-Gkiko** [M-Gkiko/data_mining_framework#1]
- [x] Initialize Git repository with `.gitignore`, `README.md`
- [x] Set up project structure (folders: `core/`, `implementations/`, `tests/`, `examples/`, `data/`)
- [x] Choose language and create `requirements.txt`
- [x] **CRITICAL:** Define ALL abstract base classes/interfaces together (**Strategy Pattern**)
  - `Dataset`: `get_data()`, `get_features()`, `get_rows()`, `shape()`
  - `DistanceMeasure`: `calculate(point1, point2)`
  - `ClusteringAlgorithm`: `fit(dataset, distance_measure, **kwargs)`, `get_labels()`
  - `QualityMeasure`: `evaluate(dataset, labels)`
- [x] Commit interfaces to `core/` folder
- [ ] Set up basic test structure
- [x] Document architecture patterns being used

###  Dataset Component 
- [x] Implement `CSVDataset` (**Strategy Pattern** - concrete implementation)
- [x] Implement `NumpyDataset` 
- [ ] Implement `SyntheticDataset`
- [ ] Download/create 3 sample datasets (Iris, Wine, Synthetic blobs)
- [ ] Basic unit tests

###  Distance Measure Component 
- [ ] Implement `EuclideanDistance` (**Strategy Pattern** - concrete implementation)
- [x] Implement `ManhattanDistance`
- [x] Implement `CosineDistance`
- [ ] Unit tests with known values

###  Clustering Component 
- [x] Set up sklearn/scipy adapters (**Adapter Pattern** - wrap external libraries)
- [x] Implement `KMeansClustering` (wrapper around sklearn.cluster.KMeans)
- [x] Start `DBSCANClustering`
- [ ] Basic tests

### Quality Measure Component
- [x] Implement `SilhouetteScore` (**Strategy Pattern** - concrete implementation)
- [x] Implement `DaviesBouldinIndex`
- [x] Start `CalinskiHarabaszScore`
- [ ] Basic tests
